ark/utils/deepcell_service_utils.py

```diff
@@ -175,11 +184,50 @@ def run_deepcell_direct(input_dir, output_dir, host='https://deepcell.org',
         }
         f.seek(0)
 
-    upload_response = requests.post(
-        upload_url,
-        timeout=timeout,
-        files=upload_fields
-    ).json()
+    # define and mount a retry instance to call the Deepcell API again if needed
+    retry_strategy = Retry(
+        total=num_retries,
+        status_forcelist=[404, 500, 502, 503, 504],
+        method_whitelist=['HEAD', 'GET', 'POST', 'PUT', 'DELETE', 'OPTIONS', 'TRACE']
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+
+    http = requests.Session()
+    http.mount('https://', adapter)
+    http.mount('http://', adapter)
+
+    total_retries = 0
+    while total_retries < num_retries:
+        # handles the case if the main endpoint can't be reached
+        print("total number of retries is {}".format(total_retries))
+        try:
+            upload_response = http.post(
+                upload_url,
+                timeout=timeout,
+                files=upload_fields
+            )
+        except RetryError as re:
+            print("Failed to reached Deepcell after %d attempts: "
```

Comment:
This is not usually a server error. We should instead pass the error message to the user so they know why this specific batch failed